src/syntax/select.ts

```diff
@@ -91,9 +97,32 @@ export type $expr_Select<
   __expr__: Expr;
   __kind__: ExpressionKind.Select;
   __modifier__: Modifier;
-} & SelectMethods<Set>;
+} & SelectMethods<Set, Expr>;
 
-interface SelectMethods<Self extends TypeSet> {
+// Base is PathNode &
+// Filter is equality &
+// Filter.args[0] is PathLeaf &
+// Filter.args[0] is unique &
+// Filter.args[0].parent.__element__ === Base.__element__
+export type inferCardinality<
```

Comment:
I think this needs to also check that the cardinality of the second arg is `Cardinality.AtMostOne`, since for example `select User filter .username = {'userA', 'userB'}` can return multiple users.

Reply:
Good call, fixed. 

I also loosened the restriction that `Base` be a `PathNode`. It now works for any Object Expression.